File: src/main/java/net/azisaba/townia/data/PermissionMatrix.java
package net.azisaba.townia.data;

public class PermissionMatrix {
    public static final String FULL_PERMS = "BDSI";
    public static final String NO_PERMS = "";

    public static boolean hasPerm(String permString, char type) {
        if (permString == null) return false;
        return permString.indexOf(type) >= 0;
    }

    public static String setPerm(String permString, char type, boolean value) {
        if (permString == null) permString = "";
        boolean has = hasPerm(permString, type);
        if (value && !has) {
            return permString + type;
        } else if (!value && has) {
            return permString.replace(String.valueOf(type), "");
        }
        return permString;
    }
}

File: src/main/java/net/azisaba/townia/data/PermissionMatrix.kt
package net.azisaba.townia.data

object PermissionMatrix {
    const val FULL_PERMS: String = "BDSI"
    const val NO_PERMS: String = ""

    fun hasPerm(permString: String?, type: Char): Boolean {
        if (permString == null) return false
        return permString.indexOf(type) >= 0
    }

    fun setPerm(permString: String?, type: Char, value: Boolean): String {
        var permString = permString
        if (permString == null) permString = ""
        val has = hasPerm(permString, type)
        if (value && !has) {
            return permString + type
        } else if (!value && has) {
            return permString.replace(type.toString(), "")
        }
        return permString
    }
}